packages/core/utils/children.js

```diff
@@ -49,3 +49,30 @@ export const filterChildren = (prop, value, children) => (
 export const filterChildrenByName = (componentName, children) => (
   filterChildren('componentName', componentName, children)
 );
+
+/**
+ * Map an array of children into an object using the `group` prop.`
+ *
+ * @param {array} children An array of React components.
+ * @returns {object} Mapping of children by group.
+ */
+export const groupChildren = (children) => {
+  const mapping = {
+    rest: [], // Array containing ungrouped children.
```

Comment:
I agree with @amyevans comment in channel that `ungrouped` would be a better name here, but overall this looks good and is really handy.

Reply:
Updated 👍